src/ast/passes/codegen_llvm.cpp

```diff
@@ -76,6 +95,334 @@ static auto getTargetMachine()
   return target;
 }
 
+struct VariableLLVM {
+  llvm::Value *value;
+  llvm::Type *type;
+};
+
+// ScopedExpr ties an SSA value to a "delete" function, that typically will end
+// the lifetime of some needed storage. You must explicitly construct a
+// ScopedExpr from either:
+// * A value only, with no associated function when out of scope.
+// * A value and associated function to run when out of scope.
+// * A value and another ScopedExpr, whose lifetime will be preserved until
+//   this value is out of scope.
+class ScopedExpr {
+public:
+  // Neither a value nor a deletion method.
+  explicit ScopedExpr() = default;
+
+  // Value only.
+  explicit ScopedExpr(Value *value) : value_(value)
+  {
+  }
+
+  // Value with an explicit deletion method.
```

Comment:
ugghhh, I really tried not to comment on these comments but they are just not valuable at ALL. If I read the larger comment at the top of the class, which is good, looking at the constructors makes perfect sense. These additional bits above each constructor are just repetitive crud.

Reply:
I know you're going to mention something about comment consistency and eventual doc blocks yada yada - so I won't block but just know my eyes hurt reading these.

Reply:
That's fine & fair, but I'm not changing anything here. This is strictly a move from the header, and it would be bad to make adjustments. Same for the comments below.

BUT as an aside, I feel like you're optimizing for the wrong thing with comments. They aren't supposed to be minimal and elegant, they are supposed to be expository & error correcting. Redundancy isn't strictly a bad thing.